Serializer/CollectionNormalizer.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+/*
+ * This file is part of the DunglasJsonLdApiBundle package.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Dunglas\JsonLdApiBundle\Serializer;
+
+use Doctrine\ORM\Tools\Pagination\Paginator;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\SerializerAwareNormalizer;
+
+/**
+ * This normalizer handle collections and paginated collections.
+ *
+ * @author Samuel ROZE <samuel.roze@gmail.com>
+ * @author Kevin Dunglas <dunglas@gmail.com>
+ */
+class CollectionNormalizer extends SerializerAwareNormalizer implements NormalizerInterface
+{
+    const HYDRA_COLLECTION = 'hydra:Collection';
+    const HYDRA_PAGED_COLLECTION = 'hydra:PagedCollection';
+
+    /**
+     * @var ResourceResolver
+     */
+    private $resourceResolver;
+
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @param ResourceResolver $resourceResolver
+     * @param RouterInterface $router
+     */
+    public function __construct(ResourceResolver $resourceResolver, RouterInterface $router)
+    {
+        $this->resourceResolver = $resourceResolver;
+        $this->router = $router;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsNormalization($data, $format = null)
+    {
+        return 'json-ld' === $format && (is_array($data) || $data instanceof \Traversable);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function normalize($object, $format = null, array $context = array())
+    {
+        $resource = $this->resourceResolver->guessResource($object, $context);
+
+        $data = [];
+        if (!isset($context['has_json_ld_context'])) {
```

Comment:
Well, this is a duplicated code from `JsonLdNormalizer`, but this is needed to be here because this normalizer is directly called when you're serializing a collection (while `JsonLdNormalizer` is not called). The best way I thought is to create a `SerializerContextInitializer` class that will "fix" the data and context in normalizers.

Injected in the constructor, the usage at the beginning of `nomalize` method of `JsonLdNormalizer` and `CollectionNormalizer` would be:
```php
public function nomalize($object, $format = null, array $context = array())
{
    $resource = $this->resourceResolver->guessResource($object, $context);
    list($data, $context) = $this->serializerContext->initialize($resource, $context);

    // ...
}
```